producer.go

```diff
@@ -72,13 +72,9 @@ func (p *segmentProducer) newWriter(cfg Config) error {
 }
 
 func (p *segmentProducer) configureSecurity(cfg Config) error {
-	// Nothing to do
-	if !cfg.tlsEnabled() && !cfg.saslEnabled() {
-		return nil
-	}
 	transport := &kafka.Transport{}
```

Comment:
We should rather use `kafka.DefaultTransport` here.

Reply:
That's a good observation. However, `kafka.DefaultTransport` returns an interface, and it provides to way to set the security configs we need. The only thing `kafka.DefaultTransport` does is to set a dial function. On the other hand, when no dial function is specified for a `kafka.Transport` a default one will be used and it's the same as in `kafka.DefaultTransport`.